packages/frontend/src/content/L2Data.ts

```diff
@@ -1,14 +1,40 @@
-export interface L2Data {
-  aggregate: ChartData
-  byProject: Record<string, ProjectData>
+import { HttpClient } from '@l2beat/common'
+import { readFile } from 'fs/promises'
+import z from 'zod'
+
+export const ChartData = z.object({
+  types: z.tuple([z.literal('date'), z.string(), z.string()]),
+  data: z.array(z.tuple([z.string(), z.number(), z.number()])),
+})
+export type ChartData = z.infer<typeof ChartData>
+
+export const ProjectData = z.object({
+  aggregate: ChartData,
+  byToken: z.record(z.string(), ChartData),
+})
+export type ProjectData = z.infer<typeof ProjectData>
+
+export const L2Data = z.object({
+  aggregate: ChartData,
+  byProject: z.record(z.string(), ProjectData),
+})
+export type L2Data = z.infer<typeof L2Data>
+
+async function getDataFromFile() {
+  const data = await readFile(`${__dirname}/../../../backend/build/data.json`, {
+    encoding: 'utf-8',
+  })
+  return JSON.parse(data)
 }
 
-export interface ProjectData {
-  aggregate: ChartData
-  byToken: Record<string, ChartData>
+async function getDataFromApi() {
+  const http = new HttpClient()
+  const response = await http.fetch('https://api.l2beat.com/api/data')
+  return response.json()
```

Comment:
You want to check `response.ok`, because if it's 400 but still valid JSON you won't get an error at this state.